pubtools/iib/iib_ops.py

```diff
@@ -248,6 +248,12 @@
     "required": False,
     "type": str,
 }
+ADD_CMD_ARGS[("--deprecation-list",)] = {
+    "group": "IIB service",
+    "help": "Comma separated list of deprecated bundles",
+    "required": False,
+    "type": str,
```

Comment:
I have seen that if I do not submit a `deprecation_list` in pub, it submits `--deprecation-list` without anything after it. Should we have a default value `" "` here to make sure that is handled?

Reply:
Example
```
[DEBUG   ] pubtools-iib-add-bundles --arch ppc64le --bundle \
[DEBUG   ]   bundle --deprecation-list \
[DEBUG   ]   --index-image quay.io/foo/index-image:v4.6 --pulp-url https://pulp-url \
[DEBUG   ]   --pulp-user admin --skip-pulp --iib-krb-ktfile /home/user/ktfile --iib-krb-principal user@EXAMPLE \
[DEBUG   ]   --overwrite-from-index --iib-server iib-server
```

Reply:
Line #538 should take care of it

Reply:
```
 ➜  pubtools-iib git:(paya_pr) ✗ pubtools-iib-add-bundles --arch ppc64le --bundle  bundle --deprecation-list --index-image quay.io/foo/index-image:v4.6         
usage: pubtools-iib-add-bundles [-h] [--pulp-url PULP_URL] [--pulp-user PULP_USER]
                                [--pulp-password PULP_PASSWORD] [--pulp-insecure]
                                [--pulp-repository PULP_REPOSITORY] [--iib-insecure] --iib-server IIB_SERVER
                                --iib-krb-principal IIB_KRB_PRINCIPAL [--iib-krb-ktfile IIB_KRB_KTFILE]
                                [--iib-cnr-token IIB_CNR_TOKEN] [--index-image INDEX_IMAGE]
                                [--binary-image BINARY_IMAGE] [--arch ARCH] [--overwrite-from-index]
                                [--overwrite-from-index-token OVERWRITE_FROM_INDEX_TOKEN] [--skip-pulp]
                                [--skip-quay] [--quay-dest-repo QUAY_DEST_REPO] [--quay-user QUAY_USER]
                                [--quay-password QUAY_PASSWORD] [--quay-remote-exec]
                                [--quay-ssh-remote-host QUAY_SSH_REMOTE_HOST]
                                [--quay-ssh-remote-host-port QUAY_SSH_REMOTE_HOST_PORT]
                                [--quay-ssh-reject-unknown-host] [--quay-ssh-username QUAY_SSH_USERNAME]
                                [--quay-ssh-password QUAY_SSH_PASSWORD]
                                [--quay-ssh-key-filename QUAY_SSH_KEY_FILENAME] [--quay-send-umb-msg]
                                [--quay-umb-url QUAY_UMB_URL] [--quay-umb-cert QUAY_UMB_CERT]
                                [--quay-umb-client-key QUAY_UMB_CLIENT_KEY]
                                [--quay-umb-ca-cert QUAY_UMB_CA_CERT] [--quay-umb-topic QUAY_UMB_TOPIC]
                                [--bundle BUNDLE] [--iib-legacy-org IIB_LEGACY_ORG]
                                [--deprecation-list DEPRECATION_LIST]
pubtools-iib-add-bundles: error: argument --deprecation-list: expected one argument
```

Reply:
It gives me this with the current code

Reply:
On second thoughts, I think this should be the correct behavior if you do not provide an empty string. I'll change Pub to never submit that arg if the deprecation_list is empty